### PR TITLE
file-based CDK: Configurable strings_can_be_null

### PR DESCRIPTION
**Pull request by girarda**
## What
* This PR address the first part of https://github.com/airbytehq/airbyte/issues/29215 by introducing the option to make strings nullable
* The default behavior is for strings to be nullable. The S3Adapter will handle backwards compatibility separately.

## How
When converting values to null:
1. If the value is not in the null_values set -> value is not None
2. If the value is in the null_values set AND its type is not "string" -> value is None
3. If the value is in the null_values set AND its type is "string" and strings_can_be_null" is True -> value is None
4.  If the value is in the null_values set AND its type is "string" and strings_can_be_null" is False -> value is not None

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py`
5. `airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/csv_parser.py`

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py [hunk 1]
```diff
@@ -43,6 +43,18 @@
                 "description": "Used during spec; allows the developer to configure the cloud provider specific options\nthat are needed when users configure a file-based source.",
                 "type": "object",
                 "properties": {
+                    "start_date": {
```
Comment (girarda) [inline airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py:46]: not 100% sure why this is being re-ordered, but properties is a map so this shouldn't matter

File: airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py [hunk 2]
```diff
@@ -64,6 +64,11 @@ class Config:
         default=[],
         description="A set of case-sensitive strings that should be interpreted as null values. For example, if the value 'NA' should be interpreted as null, enter 'NA' in this field.",
     )
+    strings_can_be_null: bool = Field(
```
Comment (clnoll) [inline airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py:67]: Should we have some validation on this that requires `null_values` to be set if `strings_can_be_none` is True?

**Review by clnoll — APPROVED** [review]
Looks great @girarda! One small question.

File: airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py (hunk 2 above)
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py:67]: I don't think we want that validation. Users should be able to configure `null_values` even if `strings_can_be_none` is False to set non-string fields to `None`

Comment (clnoll) [reply airbyte-cdk/python/airbyte_cdk/sources/file_based/config/csv_format.py:67]: Oh right, makes sense!

**Pull request merged**